frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -16,10 +17,13 @@ export const handlePipelineRunSubmit = (pipelineRun: PipelineRun) => {
     ),
   );
 };
-export const getPipelineRunData = (pipeline: Pipeline, latestRun?: PipelineRun): PipelineRun => {
-  if (!pipeline || !pipeline.metadata || !pipeline.metadata.name || !pipeline.metadata.namespace) {
+export const getPipelineRunData = (
+  pipeline: Pipeline = null,
+  latestRun?: PipelineRun,
+): PipelineRun => {
+  if (!pipeline && !latestRun) {
```

Comment:
The sub params of both of these can be null. Perhaps a deeper check? `_.has` for instance? Or you could do `_.get` and store those into variables which you use later.

Reply:
After talking with Abhishek, we're going to just get this in as-is. Likely hood of metadata missing is probably fairly low. @abhi-kn will log a task for himself to investigate after feature freeze. I think all natural flows lead to a properly created K8sResource with metadata. Wouldn't hurt to do more safe-guards in the future though.

Reply:
as agreed, created a task https://jira.coreos.com/browse/ODC-2143